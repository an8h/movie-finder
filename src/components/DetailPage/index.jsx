/* eslint-disable react/jsx-one-expression-per-line */
/* eslint-disable camelcase */
/* eslint-disable react/forbid-prop-types */

/* 
DetailPage Component: 
It is shown when the user clicks on an item  either from the categories or 
from the rearch results at Home page.
Rendered UI:
title         : The title of the clicked item
overview      : The overview of the clicked item
release_date  : The release date of the clicked item
posterToShow  : The poster image of the clicked item
Watch Video   : A button to click and go to the Watch Video Component
Close         : A button to click and go to Home Component
*/

import React, { useState } from 'react';
import { Redirect, useHistory } from 'react-router-dom';
import { PropTypes } from 'prop-types';
import { Button } from 'reactstrap';
import styles from './style.css';
import fallbackPoster from '../../img/no-image-icon-23494.png';

const DetailPage = (props) => {
  const { location } = props;
  const { state } = location;
  const { clickedItem } = state;
  const { title } = clickedItem;
  const { overview } = clickedItem;
  const { release_date } = clickedItem;
  const { posterToShow } = state;
  const history = useHistory();
  const [clicked, handleClick] = useState(false);

  return (
    <div className={styles.container}>
      <img
        className={styles.image}
        src={posterToShow}
        alt={title}
        onError={(e) => {
          e.target.src = fallbackPoster;
          e.target.style = 'object-fit: contain;';
        }}
      />
      <div className={styles.text}>
        <h2>{title}</h2>
        <p>{overview}</p>
        <h3>Release date: {release_date}</h3>
        <Button
          color="primary"
          size="lg"
          onClick={() => {
            handleClick(true);
          }}
        >
          Watch Video
        </Button>{' '}
        <Button
          color="secondary"
          size="lg"
          onClick={() => {
            history.goBack();
          }}
        >
          Close
        </Button>
      </div>
      {clicked ? <Redirect push to="/watchVideo" /> : null}
    </div>
  );
};

export default DetailPage;

DetailPage.propTypes = {
  location: PropTypes.object.isRequired,
};
