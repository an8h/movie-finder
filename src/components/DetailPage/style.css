:local(.container) {
	padding: 20px 40px 40px;
	display: flex;
}

:local(.image) {
	margin: 20px 20px 0 30px;
		width: 200px;
		object-fit: contain;
		align-self: flex-start;
}

:local(.text) {
  margin: 20px 20px 0 30px;
	flex: 1 1 auto;
}