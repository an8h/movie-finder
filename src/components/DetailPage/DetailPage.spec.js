import React from 'react';
import renderer from 'react-test-renderer';
import DetailPage from './index';

const props = {
  location: {
    pathname: '/detail',
    search: '',
    hash: '',
    key: '',
    state: {
      clicked: true,
      clickedItem: {
        title: 'this is a clicked item',
        overview: 'this is an overview of the clicked item',
        release_date: '2020-03-05',
      },
      posterToShow:
        'http://image.tmdb.org/t/p/w185/8WUVHemHFH2ZIP6NWkwlHWsyrEL.jpg',
    },
  },
};

describe('DetailPage Component', () => {
  let component;

  beforeEach(() => {
    component = renderer.create(
      <DetailPage
        location={props.location}
        clickedItem={props.location.state.clickedItem}
      />,
    );
  });

  it('should render correctly', () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
