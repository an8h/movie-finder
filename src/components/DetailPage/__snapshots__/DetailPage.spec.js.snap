// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailPage Component should render correctly 1`] = `
<div
  className="container"
>
  <img
    alt="this is a clicked item"
    className="image"
    onError={[Function]}
    src="http://image.tmdb.org/t/p/w185/8WUVHemHFH2ZIP6NWkwlHWsyrEL.jpg"
  />
  <div
    className="text"
  >
    <h2>
      this is a clicked item
    </h2>
    <p>
      this is an overview of the clicked item
    </p>
    <h3>
      Release date: 
      2020-03-05
    </h3>
    <button
      aria-label={null}
      className="btn btn-primary btn-lg"
      onClick={[Function]}
      type="button"
    >
      Watch Video
    </button>
     
    <button
      aria-label={null}
      className="btn btn-secondary btn-lg"
      onClick={[Function]}
      type="button"
    >
      Close
    </button>
  </div>
</div>
`;
